code/go/0chain.net/chaincore/transaction/entity.go

```diff
@@ -309,9 +310,14 @@ func (t *Transaction) VerifySignature(ctx context.Context) error {
 
 /*GetSignatureScheme - get the signature scheme associated with this transaction */
 func (t *Transaction) GetSignatureScheme(ctx context.Context) (encryption.SignatureScheme, error) {
-	var err error
+
 	co, err := client.GetClientFromCache(t.ClientID)
 	if err != nil {
+		err = encryption.VerifyPublicKeyClientID(t.PublicKey, t.ClientID)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
Hmm, I checked the code again, this ClientID and PublicKey should be verified when receiving the txn. I'll move the verification code there if we haven't done it yet.